oauth2/oauth2.go

```diff
@@ -257,7 +255,7 @@ func (c *Client) RequestToken(grantType, value string) (result TokenResponse, er
 	v := c.commonURLValues()
 
 	v.Set("grant_type", grantType)
-	v.Set("client_secret", c.creds.Secret)
```

Comment:
This can be a problem because it makes _AuthMethodClientSecretBasic_ AND _AuthMethodClientSecretPost_ request. However _AuthMethodClientSecretPost_ [MAY be supported](https://tools.ietf.org/html/rfc6749#section-2.3.1). IMHO if it's not supported, then these parameters should be ignored. At least Keycloak/Google/GitHub doesn't have a problem with this basic&post auth request.

Anyway, it should be removed 👍 _newAuthenticatedRequest_ will add it later eventually.